<?xml version="1.0" ?>
<launch>
	
	<!-- URDF loaded to parameter server -->
	<param name="robot_description"  textfile="$(find freetail_config)/urdf/freetail_vacuum_gripper_conf.urdf"/>
	<!-- Get TF -->
	<node pkg="robot_state_publisher" name="st_pub" type="state_publisher"/>
	<node pkg="joint_state_publisher" name="joint_state_pub" type="joint_state_publisher" args="use_gui:=true"/>

	<!-- sensor setup -->
  	<include file="$(find freetail_config)/launch/openni_try.launch"/>

	<!-- tabletop segmentation -->
	<include file="$(find freetail_object_manipulation)/launch/tabletop_segmentation.launch">
		<arg name="tabletop_segmentation_points_in" value="camera/depth_registered/points"/>
	</include>
	
	<!-- sphere segmentation -->
	<node name="test_sphere_segmentation" pkg="freetail_object_manipulation" type="test_sphere_segmentation" output="screen">
		<param name="world_frame_id" value="base_link"/>
		<param name="k_nearest" value="100"/>
		<param name="max_iterations" value="200"/>
		<param name="sac_method" value="6"/> <!-- ransac = 0 -->
		<param name="normal_distance_weight" value="0.2"/>
		<param name="distance_threshold" value="0.0005"/>
		<param name="max_radius" value="0.028"/>
		<param name="align_to_top_centroid" value="true"/>
		<param name="x_max" value="1.5"/>
		<param name="x_min" value="0.5"/>
		<param name="y_max" value="1.2"/>
		<param name="y_min" value="-1.2"/>
		<param name="z_max" value="0.5"/>
		<param name="z_min" value="-0.1"/>
	</node>
</launch>
